extensions/ql-vscode/src/model-editor/model-evaluator.ts

```diff
@@ -268,4 +290,37 @@ export class ModelEvaluator extends DisposableObject {
       }),
     );
   }
+
+  private async readAnalysisResults(
+    variantAnalysisId: number,
+    repositoryId: number,
+  ) {
+    const variantAnalysis =
+      this.variantAnalysisManager.tryGetVariantAnalysis(variantAnalysisId);
+    if (!variantAnalysis) {
+      void this.app.logger.log(
+        `Could not find variant analysis with id ${variantAnalysisId}`,
+      );
+      throw new Error(
+        "There was an error when trying to retrieve variant analysis information",
+      );
+    }
+
+    const repository = variantAnalysis.scannedRepos?.find(
+      (r) => r.repository.id === repositoryId,
+    );
+    if (!repository) {
+      void this.app.logger.log(
+        `Could not find reposiotry with id ${repositoryId} in scanned repos`,
+      );
+      throw new Error(
+        "There was an error when trying to retrieve repository information",
+      );
+    }
+
+    await this.variantAnalysisManager.loadResults(
```

Comment:
Perhaps add a comment here that this will trigger the `onRepoResultsLoaded` event handler? Otherwise it seems like the result is unused.

Reply:
Good idea! Done, let me know if it's not clear.